middleware.go

```diff
@@ -26,27 +26,78 @@ const debugRequestLogKey = "__restyDebugRequestLog"
 //_______________________________________________________________________
 
 func parseRequestURL(c *Client, r *Request) error {
-	// GitHub #103 Path Params
-	if len(r.PathParams) > 0 {
+	if l := len(c.PathParams) + len(c.RawPathParams) + len(r.PathParams) + len(r.RawPathParams); l > 0 {
+		params := make(map[string]string, l)
+
+		// GitHub #103 Path Params
 		for p, v := range r.PathParams {
-			r.URL = strings.Replace(r.URL, "{"+p+"}", url.PathEscape(v), -1)
+			if _, ok := params[p]; !ok {
```

Comment:
@SVilgelm I think `if` condition is not required for the first iteration. Your thoughts?

Reply:
the first `if` is needed to avoid the unneeded allocation of the params slice

Reply:
@SVilgelm I meant this if condition `if _, ok := params[p]; !ok {` at line 34 because `params` just got created right above; it's empty, and we need to take all the params from the request. That's why I thought a conditional check was not required.

Reply:
oh, I see

Reply:
done